phoenix-core/src/main/java/org/apache/phoenix/coprocessor/MetaDataEndpointImpl.java

```diff
@@ -1562,7 +1678,7 @@ private MetaDataMutationResult doDropTable(byte[] key, byte[] tenantId, byte[] s
             }
 
             if (tableType != PTableType.VIEW) { // Add to list of HTables to delete, unless it's a view
-                tableNamesToDelete.add(table.getName().getBytes());
+                tableNamesToDelete.add(table.getPhysicalName().getBytes());
```

Comment:
Ah, ok. Thanks for the bug fix :-). Please make sure there's a unit test that goes with it.